server/src/resolvers/User/mutation.ts

```diff
@@ -97,37 +98,35 @@ export const signInEmail = mutationField('signInEmail', {
     password: stringArg({ nullable: false }),
   },
   resolve: async (_parent, { email, password }, ctx) => {
-    const { pubsub } = ctx;
+    const { pubsub, prisma } = ctx;
 
-    const user = await ctx.prisma.user.findOne({
-      where: {
-        email,
-      },
+    const findUserWithEmail = async () => prisma.user.findOne({
+      where: { email },
     });
 
-    if (!user) {
-      throw new Error(`No user found for email: ${email}`);
-    }
-
-    const passwordValid = await validateCredential(password, user.password);
-
-    if (!passwordValid) {
-      throw new Error('Invalid password');
-    }
-
-    pubsub.publish(USER_SIGNED_IN, user);
-
-    ctx.prisma.user.update({
-      where: {
-        email,
-      },
+    const updateLastSignedIn = () => prisma.user.update({
+      where: { email },
       data: { lastSignedIn: new Date().toISOString() },
     });
 
-    return {
-      token: sign({ userId: user.id }, APP_SECRET),
-      user,
-    };
+    return await pipe(
```

Comment:
I am wondering whether if we need this `await` here since the function is already a promise and it is retrieving again inside an `async` function which also returns a promise. Let me test this out and come back.

Reply:
Yeap. We can remove `await` here.